OpenEdXMobile/src/main/java/org/edx/mobile/util/DataBindingHelperUtils.kt

```diff
@@ -114,47 +123,109 @@ class DataBindingHelperUtils {
-                textView.background = ContextCompat.getDrawable(textView.context, R.drawable.yellow_roundedbg)
-            } else {
-                setText(textView, type.getTitle())
-                when (type) {
-                    CourseDateType.VERIFIED_ONLY -> {
-                        textView.setTextColor(ContextCompat.getColor(textView.context, R.color.course_date_badge_white))
-                        textView.background = ContextCompat.getDrawable(textView.context, R.drawable.black_roundedbg)
-                        TextViewCompat.setCompoundDrawablesRelativeWithIntrinsicBounds(textView,
-                                IconDrawable(textView.context, FontAwesomeIcons.fa_lock)
-                                        .sizeRes(textView.context, R.dimen.small_icon_size)
-                                        .colorRes(textView.context, R.color.white),
-                                null, null, null
-                        )
-                    }
-                    CourseDateType.COMPLETED -> {
-                        textView.setTextColor(ContextCompat.getColor(textView.context, R.color.course_date_badge_gray))
-                        textView.background = ContextCompat.getDrawable(textView.context, R.drawable.light_silver_roundedbg)
-                    }
-                    CourseDateType.PAST_DUE -> {
-                        textView.setTextColor(ContextCompat.getColor(textView.context, R.color.course_date_badge_gray))
-                        textView.background = ContextCompat.getDrawable(textView.context, R.drawable.light_gray_roundedbg)
-                    }
-                    CourseDateType.DUE_NEXT -> {
-                        textView.setTextColor(ContextCompat.getColor(textView.context, R.color.course_date_badge_white))
-                        textView.background = ContextCompat.getDrawable(textView.context, R.drawable.dark_gray_roundedbg)
-                    }
-                    CourseDateType.NOT_YET_RELEASED -> {
-                        textView.setTextColor(ContextCompat.getColor(textView.context, R.color.course_date_badge_silver))
-                        textView.background = ContextCompat.getDrawable(textView.context, R.drawable.silver_border_transparent_roundedbg)
-                    }
-                    else -> {
-                        textView.visibility = View.INVISIBLE
+        @BindingAdapter("binding:badge", "binding:isToday", "binding:dateBlockItems", requireAll = true)
+        fun setBadge(textView: TextView, type: CourseDateType, isToday: Boolean,
+                     dateBlockItems: ArrayList<CourseDateBlock>?) {
+            if (!hasSameDateTypes(dateBlockItems) && !isToday) {
+                return
+            }
+            var courseDateType = type
+            // check dateBlockItems not null its means we can adding the badge in Parent date view
+            if (isToday && dateBlockItems != null) {
```

Comment:
Please also mention in doc why we need this check

Reply:
`dateBlockItems != null` removed.

Reply:
doc added.